app/services/task_service.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class TaskService
+  @count_per_page
+
+  def initialize(count_per_page: 10)
+    @count_per_page = count_per_page
+  end
+
+  def task_list(params: nil)
+    task_count = Task.count
+    Task
+      .offset(calc_offset(extract_page(params), @count_per_page, task_count))
+      .limit(@count_per_page)
+      .eager_load(:user)
+      .all
+  end
+
+  private
+
+  def calc_offset(page, limit, count)
+    count / limit * (page - 1)
+  end
+
+  def extract_page(params)
+    return 1 if params.nil? || params[:page].nil?
+
+    page = params[:page]
+    page.to_i
```

Comment:
There is a strange notation for that :)
```suggestion
    params&.[](:page) || 1
```

Reply:
need to convert to int )